src/core/dxf/qgsdxfexport.cpp

```diff
@@ -1042,6 +1048,7 @@ void QgsDxfExport::writeEntities()
       writeEntitiesSymbolLevels( vl );
       renderer->stopRender( ctx );
 
+      delete ctx.expressionContext().popScope();
```

Comment:
Would it be possible to use a `unique_ptr` for memory management here?

Something like

```cpp
auto scopePopper = [&ctx]( QgsExpressionContextScope* scope ) 
{
  delete ctx.popScope();
};
std::unique_ptr<QgsExpressionContextScope, decltype(scopePopper)> layerScope;
```

(untested)

Reply:
There's already ExpressionContextScopePopper in qgssymbol.cpp. This could just be moved to qgsexpressioncontext.h and renamed to QgsExpressionContextScopePopper.

Reply:
(although that one currently doesn't delete the scope - so maybe @m-kuhn's approach is better)